test/e2e/base/base-compose.yml

```diff
@@ -51,8 +80,13 @@ services:
       - e2e
     environment:
       SW_AGENT_COLLECTOR_BACKEND_SERVICES: satellite:11800
+      SW_GRPC_LOG_SERVER_HOST: satellite
+      SW_GRPC_LOG_SERVER_PORT: 11800
+      SW_COLLECTOR_GET_PROFILE_TASK_INTERVAL: 1
       SW_AGENT_NAME: e2e-service-provider
-      JAVA_OPTS: -javaagent:/skywalking/agent/skywalking-agent.jar=logging.output=CONSOLE,agent.instance_name=provider1
+      JAVA_OPTS: >-
+        -javaagent:/skywalking/agent/skywalking-agent.jar=logging.output=CONSOLE,agent.instance_name=provider1
+        -agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=5005
```

Comment:
Consider using the java agent docker image so that we don't need these

Reply:
Do you mean I don't need to declare the `JAVA_OPTS`?

Reply:
> Do you mean I don't need to declare the `JAVA_OPTS`?

Yes, our Java agent Docker image already has this to enable the SkyWalking agent

Reply:
Yet you need some env var to set the service name `provider1`

Reply:
@wu-sheng see, this is impossible for Docker / Sidecar usage. With the language agents provide Docker images and SWCK provides auto injection, we can't just have very little configuration items in the `agent.conf` (Java agent), we need all agent configurations to be able to configured by env vars.

Reply:
Those configurations valuable for testing could be added. We didn't have this infra tool before, now, the status changes.
My argue was only, we need to set the boundaries clear, which we accept, or reject.

Reply:
> Those configurations valuable for testing could be added. We didn't have this infra tool before, now, the status changes.

Not only for testing, it also applies in production deployment with our official Docker image. If we provide a configuration item for Java agent but it's impossible to configure it via env var, our Docker image makes less sense as users have to rebuild or hack the official Docker image.

> My argue was only, we need to set the boundaries clear, which we accept, or reject.
 
IMHO, if we say a behavior of agent is configurable, is should be configurable via env var

Reply:
Could we just add the full configuration when using docker images? because It could give more easy to let users use or testing.

Reply:
> > Those configurations valuable for testing could be added. We didn't have this infra tool before, now, the status changes.
> 
> 
> 
> Not only for testing, it also applies in production deployment with our official Docker image. If we provide a configuration item for Java agent but it's impossible to configure it via env var, our Docker image makes less sense as users have to rebuild or hack the official Docker image.
> 
> 
> 
> > My argue was only, we need to set the boundaries clear, which we accept, or reject.
> 
>  
> 
> IMHO, if we say a behavior of agent is configurable, is should be configurable via env var in containerized scenarios.
> 
> 

If you want this, we need all configurations like the OAP YMAL.